File: src/main/java/com/core/base/corebase/config/WebConfig.java
package com.core.base.corebase.config;

import com.core.base.corebase.repository.UserRepository;
import com.core.base.corebase.support.JwtInterceptor;
import com.core.base.corebase.support.JwtProvider;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.servlet.config.annotation.CorsRegistry;
import org.springframework.web.servlet.config.annotation.InterceptorRegistry;
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer;

@Configuration
public class WebConfig implements WebMvcConfigurer {

    private final AuthenticationFacade authenticationFacade;
    private final JwtProvider jwtProvider;
    private final UserRepository userRepository;

    public WebConfig(AuthenticationFacade authenticationFacade,
                     JwtProvider jwtProvider,
                     UserRepository userRepository) {
        this.authenticationFacade = authenticationFacade;
        this.jwtProvider = jwtProvider;
        this.userRepository = userRepository;
    }

    @Override
    public void addCorsMappings(CorsRegistry registry) {
        registry.addMapping("/**")
                .allowedOrigins("*")
                .allowedMethods("*")
                .allowedHeaders("*");
    }

    @Override
    public void addInterceptors(InterceptorRegistry registry) {
        registry.addInterceptor(new JwtInterceptor(authenticationFacade, jwtProvider, userRepository))
                .addPathPatterns("/**")
                .excludePathPatterns(
                        "/health-check", "/auth/**", "/error", "/swagger-resources/**", "/swagger-ui/**", "/v3/api-docs/**"
                );
    }
}

File: src/main/kotlin/com/core/base/corebase/config/WebConfig.kt
package com.core.base.corebase.config

import com.core.base.corebase.repository.AccountRepository
import com.core.base.corebase.repository.UserRepository
import com.core.base.corebase.support.JwtInterceptor
import com.core.base.corebase.support.JwtProvider
import org.springframework.context.annotation.Configuration
import org.springframework.web.servlet.config.annotation.CorsRegistry
import org.springframework.web.servlet.config.annotation.InterceptorRegistry
import org.springframework.web.servlet.config.annotation.WebMvcConfigurer

@Configuration
class WebConfig(
    private val authenticationFacade: AuthenticationFacade,
    private val jwtProvider: JwtProvider,
    private val accountRepository: AccountRepository,
    private val userRepository: UserRepository
) : WebMvcConfigurer {

    override fun addCorsMappings(registry: CorsRegistry) {
        registry.addMapping("/**")
            .allowedOrigins("*")
            .allowedMethods("*")
            .allowedHeaders("*")
    }

    override fun addInterceptors(registry: InterceptorRegistry) {
        registry.addInterceptor(
            JwtInterceptor(authenticationFacade, jwtProvider, accountRepository, userRepository)
        ).addPathPatterns("/**")
            .excludePathPatterns(
                "/health-check", "/auth/**", "/error", "/swagger-resources/**", "/swagger-ui/**", "/v3/api-docs/**"
            )
    }
}